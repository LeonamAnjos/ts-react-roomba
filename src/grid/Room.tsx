import React from "react";
import "../styles/Grid.css";
import Column from "./Column";
import Cell from "./Cell";

type RoomProps = {
  size: number;
};

const Room = ({ size }: RoomProps) => {
  const columns = [];
  for (let i = 0; i < size; i++) {
    const cells = [];
    for (let j = 0; j < size; j++) {
      cells.push(<Cell key={j} column={i} row={j}></Cell>);
    }

    columns.push(<Column key={i}>{cells}</Column>);
  }

  return <div className="Grid">{columns}</div>;
};

export default Room;
