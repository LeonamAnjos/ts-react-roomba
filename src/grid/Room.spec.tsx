import { render } from "@testing-library/react";
import RoombaContextProvider from "../contexts/RoombaContext";
import Room, { RoomProps } from "./Room";

describe("<Cell />", () => {
  const customRender = ({ size }: RoomProps) => {
    return render(
      <RoombaContextProvider size={4}>
        <Room size={size} />
      </RoombaContextProvider>
    );
  };
  it("SHOULD render WHEN size 2", () => {
    const { container } = customRender({ size: 2 });

    expect(container).toMatchSnapshot();
  });

  it("SHOULD render WHEN size 3", () => {
    const { container } = customRender({ size: 3 });

    expect(container).toMatchSnapshot();
  });
});
