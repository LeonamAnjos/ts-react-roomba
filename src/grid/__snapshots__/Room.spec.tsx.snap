// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> SHOULD render WHEN size 2 1`] = `
<div>
  <div
    class="Grid"
  >
    <div
      class="Column"
    >
      <div
        class="Cell"
      >
        ðŸ‘‰
      </div>
      <div
        class="Cell"
      />
    </div>
    <div
      class="Column"
    >
      <div
        class="Cell"
      />
      <div
        class="Cell"
      />
    </div>
  </div>
</div>
`;

exports[`<Cell /> SHOULD render WHEN size 3 1`] = `
<div>
  <div
    class="Grid"
  >
    <div
      class="Column"
    >
      <div
        class="Cell"
      >
        ðŸ‘‰
      </div>
      <div
        class="Cell"
      />
      <div
        class="Cell"
      />
    </div>
    <div
      class="Column"
    >
      <div
        class="Cell"
      />
      <div
        class="Cell"
      />
      <div
        class="Cell"
      />
    </div>
    <div
      class="Column"
    >
      <div
        class="Cell"
      />
      <div
        class="Cell"
      />
      <div
        class="Cell"
      />
    </div>
  </div>
</div>
`;
